flux-typeck/src/checker.rs

```diff
@@ -308,6 +319,31 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
         Ok(())
     }
 
+    fn is_stmt_nop(stmt: &Statement) -> bool {
+        match stmt.kind {
+            StatementKind::Nop => true,
+            _ => false,
+        }
+    }
+
+    fn is_bb_nop(&self, bb: BasicBlock) -> bool {
+        self.body.basic_blocks[bb]
+            .statements
+            .iter()
+            .all(|stmt| Self::is_stmt_nop(stmt))
+    }
+
+    fn is_bb_ret(&self, bb: BasicBlock) -> bool {
+        match &self.body.basic_blocks[bb].terminator {
+            None => false,
+            Some(term) => matches!(term.kind, TerminatorKind::Return),
+        }
+    }
+
+    fn is_exit(&self, bb: BasicBlock) -> bool {
+        self.is_bb_nop(bb) && self.is_bb_ret(bb)
```

Comment:
these tiny little functions don't feel idiomatic to me. I'd inline them

```rust
fn is_exit_block(&self, bb: BasicBlock) -> bool {
    let data = self.body.basic_bloks[bb];
    let is_no_op = data.statements.iter().all(Statement::is_nop);
    let is_ret = data.terminator.map(Terminator::is_return).unwrap_or(false);
    is_no_op && is_ret
}
```

Reply:
done.